src/equations/compressible_euler_2d.jl

```diff
@@ -865,6 +957,48 @@ end
 end
 
 
+# Compute numerical flux in direction of a normal vector by rotating the velocity, 
+# computing the numerical flux in x-direction, and rotating the calculated flux back.
+@inline function (numflux::FluxPlusDissipation)(u_ll, u_rr, normal_vector::SVector, equations::CompressibleEulerEquations2D)
+  norm_ = norm(normal_vector)
+
+  # cos and sin of the angle between the x-axis and the normalized normal_vector are
+  # the normalized vector's x and y coordinates respectively (see unit circle).
+  c = normal_vector[1] / norm_
+  s = normal_vector[2] / norm_
+
+  # Clockwise rotation by α
+  # Multiply with [ 1     0       0     0;
+  #                 0   cos(α)  sin(α)  0;
+  #                 0  -sin(α)  cos(α)  0;
+  #                 0     0       0     1 ]
+
+  u_ll_rotated = SVector(u_ll[1], 
+                          c * u_ll[2] + s * u_ll[3],
+                         -s * u_ll[2] + c * u_ll[3],
+                         u_ll[4])
+
+  u_rr_rotated = SVector(u_rr[1], 
+                          c * u_rr[2] + s * u_rr[3],
+                         -s * u_rr[2] + c * u_rr[3],
+                         u_rr[4])
```

Comment:
@ranocha @andrewwinters5000 I am wondering: would it make sense to define some sort of generic rotation/back rotation function that can be used such that we do not have to repeat this code for each flux? Maybe something like a `flux_rotated(numflux, u_ll, u_rr, normal_vector, equations)` function that can be implemented for each equation and that rotates the states into the x-normal direction, computes the original flux, and rotates the result back?

It might be useful to wait until we see the 3D code to actually implement this, I just wanted to hear your take on this such that I won't forget about it.

Reply:
This sounds very useful, especially since it should work the same way in 3D.

Reply:
Can we use something similar for the physical flux? E.g., along the lines of
```julia
function flux(u, normal_::AbstractVector, equations)
  normal = normalize(normal_)
  u_rotated = rotate_to_x(u, normal, equations)
  f = flux(u_rotated, 1, euqations)
  return rotate_from_x(f, normal, equations)
end
```

Reply:
We don't need to evaluate the flux in arbitrary directions for rotationally invariant equations, and I don't know where we would need it.

Reply:
OK.

Reply:
We implemented `FluxRotated`, but we are not sure if we dispatched `orientation` and `normal_vector` correctly.
We added `orientation::Integer` to the `flux` function of every equation. This should make sure that calling `central_flux` errors if there is no second `flux` function implemented for this equation that takes a vector.
So in theory, using a usual flux with equations like compressible Euler will fail, and it will be required to either use `FluxRotated` (for rotationally invariant equations) or implement a `flux` in arbitrary directions (e.g. what we did for LSA).

Reply:
I think @sloede (and I) had in mind something like adding the type constraint `orientation::Integer` to every flux function, i.e. `flux(...)` and `flux_*(...)`.

Reply:
After thinking a bit more about it - wouldn't it be more sane to add methods specialized on `normal::AbstractVector` to every `flux(...)` and `flux_*(...)`? Then, one can use the same numerical flux like `flux_chandrashekar` for both Cartesian and curvilinear meshes. This process could be simplified by creating a macro `@flux_normal` (or something named similarly) that creates the method automatically. Hmm...

Reply:
Is this something we should do in this PR, or should we discuss it in an issue and create a new PR later?

Reply:
@erik-f Yes, we should do this in a separate PR to make sure that things keep moving forward for you guys. I like the direction of @ranocha's thoughts, but it might take a few iterations until we converge on a satisfactory solution. Also, I am not sure how the 3D part will complicate this (or won't), since in 3D you might need more than the normal vectors IIRC.

Thus, @erik-f, maybe you can create an issue that links back to this discussion, and summarize the current state of development. Then we won't forget it and can discuss it independent of this PR.

Reply:
After the issue has been created, I think you can resolve this conversation.

Reply:
#506